### PR TITLE
Added backgroundDecoration property to Scaffold.

### LINKED ISSUE #109223: [proposal] add "backgroundDecoration" to Scaffold
## Use case

A common way decorate the background of a [scaffold](https://api.flutter.dev/flutter/material/Scaffold-class.html), 
beyond the [backgroundColor](https://api.flutter.dev/flutter/material/Scaffold/backgroundColor.html) property (maybe an image or a gradient), 
is to add a [stack](https://api.flutter.dev/flutter/widgets/Stack-class.html) in the [body](https://api.flutter.dev/flutter/material/Scaffold/body.html).

By definition the scaffold body is `The primary content of the scaffold.`, 
which is not the use case for a background decoration.

## Proposal

Adding a `Widget` or `BoxDecoration` parameter `backgroundDecoration` 
could make the usage of the scaffold **cleaner** 
and better separate the content and "un-clutter" the body from the underlying decoration.

```dart
     Scaffold(
        backgroundDecoration: BoxDecoration(
          image:  DecorationImage( 
            //...
          ),
        ),
        // body: Stack(
        body: ActualContent( //...

```

### PR DESCRIPTION
**Pull request by chinmoy12c**
This PR adds backgroundDecoration to Scaffold.

Fixes: #109223

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -2984,6 +2994,26 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
                   snackBarWidth: snackBarWidth,
                 ),
                 children: children,
+              ) : DecoratedBox(
+                decoration: widget.backgroundDecoration!,
+                child: CustomMultiChildLayout(
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/scaffold.dart:2999]: I think here we can extract the `CustomMultiChildLayout` and its children out into a Widget and make it easier to read.

Comment (chinmoy12c) [reply packages/flutter/lib/src/material/scaffold.dart:2999]: Hey, when I extract out the `CustomMultiChildLayout` into a widget, some of the tests fail with this error message:

```
The following assertion was thrown while running async test code:
Golden "material.bottom_app_bar.custom_shape.2.png": Pixel test failed, 1.86% diff detected.
```

Any idea about this :thinking:

**Comment by Hixie** [comment]
@chinmoy12c thanks for your contribution! It looks like some tests are still failing. Before we can land this we will need to make sure all the tests pass.

I've also asked @HansMuller to take a quick look and verify that this is a feature we would want to add to the API.

Thanks!

**Comment by HansMuller** [comment]
I realize that we're not setting any land speed records in reviewing this PR. Sorry about that.

Although it covers I'm not sure that a ShapeDecoration parameter is the best way to factor this feature since it binds background color, border, and shape together. That makes it difficult to say: just change the shape but leave the background color as it is, or just add a border and use the default shape.  It also complicates the semantics of the existing backgroundColor parameter.

**Comment by chinmoy12c** [comment]
> I realize that we're not setting any land speed records in reviewing this PR. Sorry about that.
> 
> Although it covers I'm not sure that a ShapeDecoration parameter is the best way to factor this feature since it binds background color, border, and shape together. That makes it difficult to say: just change the shape but leave the background color as it is, or just add a border and use the default shape. It also complicates the semantics of the existing backgroundColor parameter.

In your opinion, what would be the best way to integrate this then without overcomplicating the existing semantics?

**Comment by HansMuller** [comment]
Maybe an optional background widget parameter that's stacked behind the scaffold body and examples showing how to use it to show a background image or a gradient? We'd also have to advise developers about what it means to also specify backgroundColor.

**Comment by Hixie** [comment]
@chinmoy12c Thanks for your contributions! Is this PR something you are still interested in working on?

**Comment by chinmoy12c** [comment]
> @chinmoy12c Thanks for your contributions! Is this PR something you are still interested in working on?

Hey, sorry I missed this one. I'll work on this as soon as I find some free time.

**Comment by goderbauer** [comment]
(triage) I am going to close this one for now since it is becoming rather stale. If you find the time to work on this again, please do not hesitate to reopen it.

**Pull request closed without merging**